### PR TITLE
Fileshare driver and api modification for create and delete

### PR DESCRIPTION
**Pull request by PravinRanjan10**
**What this PR does / why we need it**:
It support driver code. Also some enhancement in API.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 1]
```diff
@@ -1,50 +1,214 @@
...
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+//    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+//    License for the specific language governing permissions and limitations
+//    under the License.
```
Comment (wisererik) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: license clarify is not correct

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 2]
```diff
@@ -1,50 +1,214 @@
...
+	//get volume group
+	var vg = opt.GetPoolName()
+	// Crete a directory to mount
+	var dirName = path.Join("/var/", name)
+	// create a fileshare path
+	var lvPath = path.Join("/dev", vg, name)
+
+	if err := d.cli.CreateDirectory(dirName); err != nil {
+		log.Error("Failed to create a directory:", err)
+		return nil, err
```
Comment (wisererik) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: lower case for error log

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 3]
```diff
@@ -1,50 +1,214 @@
...
+	if err = d.cli.CreateVolume(name, vg, opt.GetSize()); err != nil {
+		return
+	}
+	// remove created volume if got error
+	defer func() {
+		// using return value as the error flag
+		if fshare == nil {
+			if err := d.cli.Delete(name, vg); err != nil {
+				log.Error("Failed to remove volume fileshare:", err)
+			}
```
Comment (wisererik) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: same here

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 4]
```diff
@@ -1,50 +1,214 @@
...
+		}
+	}()
+
+	// Crete fileshare on this path
+	if err := d.cli.CreateFileShare(lvPath); err != nil {
+		log.Error("Failed to create filesystem logic volume:", err)
+		return nil, err
+	}
```
Comment (wisererik) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: same here

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 5]
```diff
@@ -1,50 +1,214 @@
...
+
+	// mount the volume to directory
+	if err := d.cli.Mount(lvPath, dirName); err != nil {
+		log.Error("Failed to mount a directory:", err)
+		return nil, err
```
Comment (wisererik) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: same here

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 6]
```diff
@@ -1,50 +1,214 @@
...
+	}
+	// get export location of fileshare
+	var location []string
+	location = []string{d.cli.GetExportLocation(name, server)}
+	fmt.Println("locations :", location)
+	if len(location) == 0 {
+		log.Error("Failed to get Export location:", err)
+		return nil, err
```
Comment (wisererik) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: same here

Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/nfs.go]: done!!

File: contrib/drivers/filesharedrivers/nfs/nfs.go (hunk 5 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/nfs.go]: done!!

File: contrib/drivers/filesharedrivers/nfs/nfs.go (hunk 4 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/nfs.go]: done!!

File: contrib/drivers/filesharedrivers/nfs/nfs.go (hunk 3 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/nfs.go]: done!!

File: contrib/drivers/filesharedrivers/nfs/nfs.go (hunk 2 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/nfs.go]: done!!

File: contrib/drivers/filesharedrivers/nfs/nfs.go (hunk 1 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/nfs.go]: done!!

File: contrib/drivers/filesharedrivers/drivers.go [hunk 7]
```diff
@@ -43,23 +45,25 @@ type FileShareDriver interface {
 
 // Init
 func Init(resourceType string) FileShareDriver {
+	fmt.Println("resourceType ...", resourceType)
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/drivers.go]: Please remove the debug message, or use log.Infof.V(5) instead

File: contrib/drivers/filesharedrivers/nfs/cli.go [hunk 8]
```diff
@@ -0,0 +1,237 @@
+//    Copyright 2019 The OpenSDS Authors.
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/nfs/cli.go]: Please update the copyright style.

File: contrib/drivers/filesharedrivers/nfs/nfs.go [hunk 9]
```diff
@@ -15,36 +15,191 @@
...
+
+const (
+	KLvPath        = "lvPath"
+	KLvsPath       = "lvsPath"
+	KFileshareName = "nfsFileshareName"
+	KFileshareID   = "nfsFileshareID"
 )
 
 // NFSDriver
 type NFSDriver struct{}
```
Comment (leonwanghui) [inline contrib/drivers/filesharedrivers/nfs/nfs.go]: Please remove this struct if not used.

File: pkg/model/proto/model.pb.go [hunk 10]
```diff
@@ -6,11 +6,12 @@ package proto
 import (
 	context "context"
 	fmt "fmt"
+	math "math"
```
Comment (leonwanghui) [inline pkg/model/proto/model.pb.go]: Did u update model.proto?

File: contrib/drivers/filesharedrivers/nfs/nfs.go (hunk 9 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/nfs.go]: done!!

File: contrib/drivers/filesharedrivers/drivers.go (hunk 7 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/drivers.go]: done!!

File: contrib/drivers/filesharedrivers/nfs/cli.go (hunk 8 above)
Comment (PravinRanjan10) [reply contrib/drivers/filesharedrivers/nfs/cli.go]: //    Copyright (c) 2019 The OpenSDS Authors., Ltd. All Rights Reserved.

I hope above copyright style is correct

Comment (leonwanghui) [reply contrib/drivers/filesharedrivers/nfs/cli.go]: No, it should be `// Copyright 2019 The OpenSDS Authors.`, only one blank space required.

File: contrib/drivers/filesharedrivers/nfs/cli.go [hunk 11]
```diff
@@ -0,0 +1,237 @@
+//    Copyright (c) 2019 The OpenSDS Authors., Ltd. All Rights Reserved.
+//
```
Comment (wisererik) [inline contrib/drivers/filesharedrivers/nfs/cli.go]: Copyright (c) 2019 The OpenSDS Authors., Ltd. All Rights Reserved. -> Copyright 2019 The OpenSDS Authors.

File: pkg/model/proto/model.pb.go (hunk 10 above)
Comment (PravinRanjan10) [reply pkg/model/proto/model.pb.go]: updated the model.proto
Thanks

**Pull request merged**